### ISSUE TITLE
Organizations / Teams not working properly

### ISSUE THREAD

**Issue by mfnalex** [wekan/wekan#4178]
Hi, I encountered a very strange problem. Imagine you have a board that was assigned to an organization, or a team. All existing org/team members can now access it. However, when a new user registers, and I now assign the existing org/team to it, they are not able to edit the board. I have to add them manually. This can't be correct, right?

Another question on the side: what's the difference between organizations and teams anyway?

Thanks!

WeKan version: 5.78.0

**Comment by Emile840** [comment]
Hi @mfnalex,

You are right, when adding a new user to a team already attached to a board, this new user is not taken into account automatically (the functionality is not yet done). But if you delete this team and add it back to this board, the new user will be taken into account.

For me, there is no difference between organizations and teams as they are construct today, I think, we have to define first the relationship between an organization and a team functionally speaking and then make technical changes corresponding (@xet7 what do you think about it).

**Comment by mfnalex** [comment]
Thanks for your reply :) This was driving me insane lol, it was always the same person who didn't have a working account. It would be nice if this could be fixed because having to manually remove and readd a team from EVERY board basically makes using teams useless for me.

**Comment by xet7** [comment]
@Emile840 

Yes adding this would be very helpful.

**Comment by helioguardabaxo** [comment]
Hi, guys!

Would be interesting if a team had a organization for department's board.
But if was a multi department board, would stay in a parent org.
Seems like this:
![image](https://user-images.githubusercontent.com/24301085/143597709-bc7f5498-997d-4bdb-8dad-d8c90643b9a9.png)

- One team can have one org (parent or child org)
- One org can have multiples teams and child orgs
- One person should have a parent org and one child org
- One person can have multiples teams
What do you thing?
It's confused?

**Comment by xet7** [comment]
@helioguardabaxo 

That kind of Teams/Orgs is already possible with current WeKan, by adding those persons to those Teams/Orgs. Current way to manage that at Admin Panel is simple enough.

But even thinking about sometime having that kind of random org chart creation feature seems a little scary. Maybe keep those org charts outside of WeKan?

**Comment by xet7** [comment]
Hmm, just a correction, I think it's not scary after all. I presume creating that kind of org chart could most likely be done by just looking from database about who is at what team/org etc and showing the result in HTML page.

**Comment by helioguardabaxo** [comment]
I hadn't thought about graphics, but it's a good idea.
The main idea is to create a hierarchy between orgs to make searching for teams easier. In my institution we have 4 hierarchical levels (pro-rectories, departments, coordinations and sectors). Organizing them is essential. Maybe it's the basis for an improvement in "My boards" with board filters by teams and orgs.

**Comment by xet7** [comment]
@helioguardabaxo 

I think you mean `All Boards` page? One possible way would be Organization/Teams/etc folders, like at Android/iOS you can drop app icon on top of another icon, so those will be in same folder. Or alternatively, have those folder structures pre-made with org structure. [UCS](https://github.com/wekan/wekan/wiki/UCS) did it that way, and WeKan has beginnings of similar but without any subfolders:

![Univention_Management_Console](https://user-images.githubusercontent.com/15545/143617375-647a64cd-842a-42ad-a0fc-b8a4937bb37d.png)

**Comment by xet7** [comment]
Also similarly could be grouping at Admin Panel.

**Comment by helioguardabaxo** [comment]
Oh, yes! All boards...
A panel similar this could be very beautiful and useful.
In my case folder structures pre-made with org structure is better for collaboration boards.
For personal boards, could have a "Personal" folder with a possibility to create sub folders.

**Comment by xet7** [comment]
Yes, anyone can try to make some changes related to this and send PR.

**Comment by helioguardabaxo** [comment]
@xet7 
Just for curiosity. I saw in other issue that you are working in a "Big picture".
What is it mean?

**Comment by xet7** [comment]
@helioguardabaxo 

It means:
- Looking at current and at Roadmap upcoming WeKan features, thinking how they fit together
- Drawing big picture (about WeKan architecture) with Inkscape, adding all current and future features, fixes, grouping all together, seeing what is related
- Figuring out can I for example fix a group of bugs or open issues by making some changes
- Thinking how to improve WeKan in general

WeKan did exists already when I started maintaining Wekan at 2016-12. After that it has been just adding features and fixes. This kind of big picture does not exist yet, I did not do any of original architecture and design.

**Comment by helioguardabaxo** [comment]
Thanks for the info.
It's a important, complex and big work.

**Comment by chrisi51** [comment]
> Hi, I encountered a very strange problem. Imagine you have a board that was assigned to an organization, or a team. All existing org/team members can now access it. However, when a new user registers, and I now assign the existing org/team to it, they are not able to edit the board. I have to add them manually. This can't be correct, right?
> 
> Another question on the side: what's the difference between organizations and teams anyway?
> 
> Thanks!
> 
> WeKan version: 5.78.0

the original issue is still open, correct?
i recently got into the trap to demote all boardadmins by removing and re-adding the organization. So you have to keep in mind to put the admin rights back to the board admins afterwards.

**Comment by DarthJahus** [comment]
Same problem. I've had to remote/add team so new team users had edit access to the board.

Note that only edit was missing before applying the workaround. Users could see the board and cards, but not create or edit anything.

**Comment by chrisi51** [comment]
i've just asked chatGPT if it was a way to change the check for active user and add a check for team and organisation at https://github.com/wekan/wekan/blob/d04c5c9846f7beacb2144f4b91a332af8b724d60/models/boards.js#L742 like that:
result is something like this. 
```
isActiveMember(userId) {
  if (!userId) {
    return false;
  }
  
  const isActiveMember = this.members.find(
    member => member.userId === userId && member.isActive,
  );

  if (isActiveMember) {
    return true;
  }

  // Check if the user is part of any active team in this board
  const userTeams = Teams.find({ members: userId, isActive: true }).fetch();
  const activeTeams = this.teams.filter(team => team.isActive);
  for (let team of activeTeams) {
    if (userTeams.some(userTeam => userTeam._id === team.teamId)) {
      return true;
    }
  }

  // Check if the user is part of any active organization in this board
  const userOrgs = Organizations.find({ members: userId, isActive: true }).fetch();
  const activeOrgs = this.orgs.filter(org => org.isActive);
  for (let org of activeOrgs) {
    if (userOrgs.some(userOrg => userOrg._id === org.orgId)) {
      return true;
    }
  }

  return false;
}
```

If i understand correctly, the current way is to add ppl from teams or orgs to the regular userlist and than checking those user for permissions instead of checking for teams and orgs.  So would that help, to check for teams and orgs too to make changes at teams or orgs instantly having effect?
